### PR TITLE
Add centrality metrics and graph summary methods

### PR DESCRIPTION
**Pull request by zainkhatri**
This PR adds methods to measure node importance and summarize graph properties in fmdtools models.

## New methods

**Centrality metrics** - Identify important nodes:
- `calc_betweenness_centrality()` - Find nodes that connect different parts of the system
- `calc_closeness_centrality()` - Find nodes that are most central to the network
- `calc_eigenvector_centrality()` - Find nodes connected to other important nodes
- `plot_centrality()` - Visualize node importance with color coding

**Graph summary** - Get key statistics:
- `summary()` - Returns metrics like density, connectivity, path length, and modularity

**Graph comparison** - Track changes:
- `compare_with()` - Compare two graphs to see what nodes and edges changed

## Why this helps

Engineers can now answer questions like:
- Which components are most critical to the system?
- How connected is my system architecture?
- How did the graph structure change after a fault?

### CODE REVIEW

File: fmdtools/analyze/graph/base.py [hunk 1]
```diff
@@ -978,6 +987,340 @@ def sff_model(self, endtime=5, pi=.1, pr=.1,
...
+
+        # Only compute ASPL if graph is connected
+        if is_connected:
+            summary_dict['aspl'] = self.calc_aspl()
+        else:
+            summary_dict['aspl'] = None
+
+        summary_dict['modularity'] = self.calc_modularity()
+
+        return summary_dict
```
Comment (hulsed) [inline fmdtools/analyze/graph/base.py]: What is the use-case for this? Are we using this anywhere?

**Review by hulsed — CHANGES_REQUESTED** [review]
See comments inside. I think to show the value of this stuff you should incorporate/demonstrate usage in a notebook, such as:
- https://nasa.github.io/fmdtools/examples/rover/Model_Structure_Visualization_Tutorial.html, which could use more description of the usage of visualization of graph metrics
- https://nasa.github.io/fmdtools/examples/pump/Pump_Example_Notebook.html, which already shows this

File: fmdtools/analyze/graph/base.py [hunk 2]
```diff
@@ -978,6 +987,340 @@ def sff_model(self, endtime=5, pi=.1, pr=.1,
...
+            Dictionary of nodes with closeness centrality as values.
+
+        Examples
+        --------
+        >>> graph = Graph(ex_nxgraph)
+        >>> centrality = graph.calc_closeness_centrality()
+        >>> type(centrality)
+        <class 'dict'>
+        """
+        return nx.closeness_centrality(self.g)
```
[the remark this answers is not in the document]
Comment (zainkhatri) [reply fmdtools/analyze/graph/base.py]: Removed all 3 pass through centrality methods. Users can call NetworkX directly (e.g., nx.betweenness_centrality(mg.g)). Also updated find_vulnerable_nodes() to call NetworkX directly.

File: fmdtools/analyze/graph/base.py [hunk 3]
```diff
@@ -978,6 +987,340 @@ def sff_model(self, endtime=5, pi=.1, pr=.1,
...
+                # If all else fails, return zeros
+                return {node: 0.0 for node in self.g.nodes()}
+
+    def plot_centrality(self, metric='betweenness', title='', cmap='YlOrRd', **kwargs):
+        """
+        Plot nodes colored by centrality metric.
+
+        Visualizes the graph with nodes grouped by their centrality scores into
+        quartiles (low, medium_low, medium_high, high).
+
```
[the remark this answers is not in the document]
Comment (zainkhatri) [reply fmdtools/analyze/graph/base.py]: Refactored plot_centrality() to accept:

- metric_func: any callable that returns {node: value}
- quartiles: customizable percentile boundaries (default [0, 25, 50, 75])
- quartile_styles: optional dict for custom styling per quartile

Still accepts string shortcuts for backward compatibility.

File: fmdtools/analyze/graph/base.py (hunk 1 above)
Comment (zainkhatri) [reply fmdtools/analyze/graph/base.py]: Added demonstrations in Graph_Metrics_Tutorial.ipynb showing use cases:

- Model documentation (quick architecture reports)
- Design comparison (comparing model variants)
- Version tracking (recording structural evolution)

Unlike pass throughs, this aggregates multiple NetworkX calls and handles edge cases

File: tests/test_graph.py [hunk 4]
```diff
@@ -91,6 +91,72 @@ def test_result_from_plot(self):
...
+
+    def test_plot_centrality(self):
+        """Test centrality visualization method."""
+        mg = FunctionArchitectureGraph(self.mdl)
+        mg.set_pos(auto='spring')
+        
+        # Test different centrality metrics
+        for metric in ['betweenness', 'closeness', 'degree']:
+            fig = mg.plot_centrality(metric=metric)
+            self.assertIsNotNone(fig)
```
[the remark this answers is not in the document]
Comment (zainkhatri) [reply tests/test_graph.py]: Created Graph_Metrics_Tutorial.ipynb demonstrating all features with visual outputs:

- Centrality visualization (betweenness, closeness, degree, custom metrics)
- Summary statistics
- Model comparison examples

File: tests/test_graph.py [hunk 5]
```diff
@@ -91,6 +91,72 @@ def test_result_from_plot(self):
...
+        expected_keys = ['num_nodes', 'num_edges', 'density', 'is_connected',
+                        'num_components', 'avg_degree', 'modularity']
+        for key in expected_keys:
+            self.assertIn(key, summary)
+        
+        # Check types
+        self.assertIsInstance(summary['num_nodes'], int)
+        self.assertIsInstance(summary['num_edges'], int)
+        self.assertIsInstance(summary['density'], float)
+        self.assertIsInstance(summary['is_connected'], bool)
```
[the remark this answers is not in the document]
Comment (zainkhatri) [reply tests/test_graph.py]: Fair. I enhanced tests to verify actual values and constraints (e.g., density between 0-1, positive counts, ASPL handling for connected vs disconnected graphs).

**Review by hulsed — COMMENTED** [review]
It seems like there are more methods in here than the previous PR? What functionality are you adding with this change?
